File: src/net/masterthought/dlanguage/highlighting/DHighlighterFactory.java
package net.masterthought.dlanguage.highlighting;

import com.intellij.openapi.fileTypes.SyntaxHighlighter;
import com.intellij.openapi.fileTypes.SyntaxHighlighterFactory;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;
import net.masterthought.dlanguage.highlighting.DHighlighter;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;


/**
 *
 */
public class DHighlighterFactory extends SyntaxHighlighterFactory {

    @NotNull
    @Override
    public SyntaxHighlighter getSyntaxHighlighter(@Nullable final Project project,
                                                  @Nullable final VirtualFile virtualFile)
    {
        return new DHighlighter();
    }
}

File: src/main/kotlin/io/github/intellij/dlanguage/highlighting/DHighlighterFactory.kt
package io.github.intellij.dlanguage.highlighting

import com.intellij.openapi.fileTypes.SyntaxHighlighter
import com.intellij.openapi.fileTypes.SyntaxHighlighterFactory
import com.intellij.openapi.project.Project
import com.intellij.openapi.vfs.VirtualFile

class DHighlighterFactory : SyntaxHighlighterFactory() {
    override fun getSyntaxHighlighter(project: Project?, virtualFile: VirtualFile?): SyntaxHighlighter =
        DHighlighter()
}
